### PR TITLE
TER-77 repeated order by

### PR DESCRIPTION
**Pull request by martinjunger**
Jira: TER-77
driver-common: https://github.com/keboola/php-storage-driver-common/pull/19
connection: https://github.com/keboola/connection/pull/4097

Composer dev branches:
```
$ dcrr dev composer update keboola/storage-driver-common

        "keboola/storage-driver-common": "dev-TER-77-repeated-orderBy as 1.4.0",
```

### CODE REVIEW

File: src/Handler/Table/Preview/PreviewTableHandler.php [hunk 1]
```diff
@@ -90,10 +90,17 @@ public function __invoke(
                 assert(is_numeric($command->getChangeUntil()), 'PreviewTableCommand.changeUntil must be numeric timestamp');
             }
 
-            if ($command->hasOrderBy() && $command->getOrderBy()) {
-                /** @var PreviewTableOrderBy $orderBy */
-                $orderBy = $command->getOrderBy();
-                assert($orderBy->getColumnName() !== '', 'PreviewTableCommand.orderBy.columnName is required');
+            if ($command->getOrderBy()->count()) {
```
Comment (jirkasemmler) [inline src/Handler/Table/Preview/PreviewTableHandler.php:93]: je tady potreba ten if ? getOrderBy() je `array|RepeatedField` takze se to pripadne neiteruje

File: tests/functional/UseCase/Table/PreviewTableTest.php [hunk 2]
```diff
@@ -769,8 +820,7 @@ private function dropTable(string $databaseName, string $tableName): void
      *     changedUntil?: string,
      *     fulltextSearch?: string,
      *     whereFilters?: TableWhereFilter[],
-     *     orderBy?: array<string, int>,
-     *     orderByDataType?: int,
+     *     orderBy?: PreviewTableOrderBy[],
```
Comment (jirkasemmler) [inline tests/functional/UseCase/Table/PreviewTableTest.php:823]: to by nemelo byt nullable ne?

**Review by jirkasemmler — CHANGES_REQUESTED** [review]

File: tests/functional/UseCase/Table/PreviewTableTest.php (hunk 2 above)
Comment (martinjunger) [reply tests/functional/UseCase/Table/PreviewTableTest.php:823]: to neni nullalble, ale ze tam ten klic muze chybet. To je vlastni pole z providera, ktery se v ty metode sklada do objektu...

Comment (martinjunger) [reply tests/functional/UseCase/Table/PreviewTableTest.php:823]: souvisi s tim `isset($commandInput['orderBy']))` nize:
```php
        if (isset($commandInput['orderBy'])) {
            $orderBy = new RepeatedField(GPBType::MESSAGE, PreviewTableOrderBy::class);
            foreach ($commandInput['orderBy'] as $orderByOrig) {
                $orderBy[] = $orderByOrig;
            }
            $command->setOrderBy($orderBy);
        }
```

File: src/Handler/Table/Preview/PreviewTableHandler.php (hunk 1 above)
Comment (martinjunger) [reply src/Handler/Table/Preview/PreviewTableHandler.php:93]: fixnuto

**Review by jirkasemmler — APPROVED** [review]

**Comment by martinjunger** [comment]
Pridal jsem jeste presun `PreviewTableOrderBy` do sdileneho `ImportExportShared.OrderBy` - vyuzije se pri exportu apod.

**Pull request merged**